wordpress-mobile/WordPress-iOS
# [Modal Layout Picker] "Create Page" button functionality

**Fixes** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2446

## To test:
<details>
<summary><strong>To disable or enable the development version of Modal Layout Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker will be disabled.
    - From the site page:
        - Click on your Gravatar.
        - Click on App Settings.
        - Click on Debug.
        - Toggle "Gutenberg Modal Layout Picker" to enable or disable the picker
            <img width=300 src="https://user-images.githubusercontent.com/3384451/90816133-c8b10580-e2f9-11ea-8584-6bbfd7e523e2.gif" />
</details>

When enabled, the Layout Picker should show when creating a new page from My Site and from the Page list.

🗒️ The layouts that are provided right now do contain a lot of unsupported blocks. This will be filtered out by a later PR.

### From My Site
1. On the My Site Page
1. Click the FAB icon 
1. Select Site Page
1. Choose a Layout
1. Select Create Page

**Expect** to see the Editor loaded with the contents of your selected block

### From Pages
1. On the My Site Page
1. Open the Page list
1. Select the ➕ icon
1. Choose a Layout
1. Select Create Page

**Expect** to see the Editor loaded with the contents of your selected block

### From Deep Link 
1. Make sure the test build is the only build installed
   - This is to make sure we're following the directions in https://github.com/wordpress-mobile/WordPress-iOS/pull/13707
     > All other copies of the app must be uninstalled since I believe it is random which copy of the app is chosen to open the URL
1. From another app (I used notes but sending a text in messages should also work) set up the link `https://apps.wordpress.com/get#/page`
1. Open the link by clicking on it
    - **Expect** to see the Layout picker
1. Choose a Layout
1. Select Create Page

**Expect** to see the Editor loaded with the contents of your selected block

### From Deep Link with specific domain
1. Make sure the test build is the only build installed
   - This is to make sure we're following the directions in https://github.com/wordpress-mobile/WordPress-iOS/pull/13707
     > All other copies of the app must be uninstalled since I believe it is random which copy of the app is chosen to open the URL
1. From another app (I used notes but sending a text in messages should also work) set up the link `https://apps.wordpress.com/get#/page/yourAwesomeDomain.com`
    - Make sure you replace `yourAwesomeDomain.com` with a domain linked to your app 😉 
1. Open the link by clicking on it
    - **Expect** to see the Layout picker
1. Choose a Layout
1. Select Create Page

**Expect** to see the Editor loaded with the contents of your selected block

### From Deep Link with content
1. Make sure the test build is the only build installed
   - This is to make sure we're following the directions in https://github.com/wordpress-mobile/WordPress-iOS/pull/13707
     > All other copies of the app must be uninstalled since I believe it is random which copy of the app is chosen to open the URL
1. From another app (I used notes but sending a text in messages should also work) set up the link `wpdebug://newpage?content=test`
1. Open the link by clicking on it

**Expect** to see the Editor loaded with the content from your deep link. Note this seems to be loading the classic editor even if Gutenberg is configured for your domain.

## Screenshots:

| My Sites | Pages |
| --- | --- |
| ![CreatePage](https://user-images.githubusercontent.com/3384451/91208486-47ca8300-e6d8-11ea-8897-64b200d3e992.gif) | ![CreatePage_pages](https://user-images.githubusercontent.com/3384451/91209428-b9ef9780-e6d9-11ea-883a-5f28c83798de.gif) |

## Additional Context: 

This is a continuation of the work done in https://github.com/wordpress-mobile/WordPress-iOS/pull/14726 and will be followed up by other tickets as part of [Modal Layout Picker Project](https://github.com/wordpress-mobile/gutenberg-mobile/issues?q=is%3Aissue+is%3Aopen+%5BModal+Layout+Picker%5D)

There are some design considerations that are still being figured out. You can see them documented here: https://github.com/wordpress-mobile/gutenberg-mobile/issues/2563 

## PR submission checklist:
